script/src/verify.rs

```diff
@@ -112,6 +116,22 @@ where
         self.debug_printer = Arc::new(func);
     }
 
+    /// Sets a new syscall generator for special usages
+    pub fn set_syscall_generator<F>(&mut self, func: F)
```

Comment:
this fn can be simplified if applied previous review comment
```
    pub fn set_syscall_generator(&mut self, func: SyscallGenerator<DL>) {
        self.syscall_generator = func;
    }
```

Reply:
The API is designed this way to mimic the above set_debug_printer method.

Reply:
Also see the above comment, function pointer is not enough for us. However, when `SyscallGenerator` is defined as `Arc<Fn(...) -> ...>`, we could indeed write:

```
pub fn set_syscall_generator(&mut self, func: SyscallGenerator<DL>) {     
    self.syscall_generator = func;
}
```

The current API is written so as to mimic existing `set_debug_printer` method. I do believe they should behave similarly.